src/types/announcement.py

```diff
@@ -0,0 +1,12 @@
+from src.types.sized_bytes import bytes32
+from dataclasses import dataclass
+from src.util.hash import std_hash
+
+
+@dataclass(frozen=True)
+class Announcement:
+    parent_coin_info: bytes32
+    message: bytes
+
+    def name(self) -> bytes32:
+        return std_hash(bytes(self.parent_coin_info + self.message))
```

Comment:
Kinda gross, probably `std_hash` should be tweaked to allow multiple arguments since the underlying hasher allows it.